### PR TITLE
Bamboo integration tests using Python frontend

### PR DESCRIPTION
**Pull request by timmoon10**
This PR adds two integration tests using the Python frontend:

- ResNet-18; 5 epochs of ~0.1% of ImageNet
- ResNet-50; 5 epochs of ImageNet (weekly build only, disabled on Catalyst and Corona)

It takes ~2 min to run ResNet-18 on Catalyst and ~2 hours to run ResNet-50 on Pascal. Unless we want to increase our time budget (~15 min for a nightly build, ~3 hours for a weekly build) or make these tests smaller, there's not any room for other big integration tests. With this in mind, I've taken the liberty of removing all of the old prototext-based integration tests.

This is ready for review. I just need to run the tests to see what runtimes and accuracies we get.

### CODE REVIEW

File: bamboo/integration_tests/common_code.py [hunk 1]
```diff
@@ -1,269 +0,0 @@
-import sys
```
Comment (forsyth2) [inline bamboo/integration_tests/common_code.py]: We don't want to keep any of the old integration tests? These two new tests will be sufficient?

File: bamboo/integration_tests/test_integration_resnet18.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    message = lbann.lbann_pb2.LbannPB()
+    with open(pb_file, 'r') as f:
+        google.protobuf.text_format.Merge(f.read(), message)
+    message = message.data_reader
+
+    # Only use data reader for training set
+    del message.reader[1:]
+
+    # We want to train with 1280 out of 1.28M samples in ImageNet-1K
+    message.reader[0].percent_of_data_to_use = imagenet_fraction
```
[the remark this answers is not in the document]
Comment (timmoon10) [reply bamboo/integration_tests/test_integration_resnet18.py]: I think this should be controlled by the test. If you change the fraction, you also need to change all the expected metrics and runtimes.

File: bamboo/integration_tests/common_code.py (hunk 1 above)
Comment (timmoon10) [reply bamboo/integration_tests/common_code.py]: I think ResNet-18 and ResNet-50 do a good job validating our image classification functionality and they use up all the available time. We'll need to make these tests smaller if we also want to test autoencoders or non-vision models.

File: bamboo/integration_tests/test_integration_resnet18.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+def construct_model(lbann):
+    """Construct LBANN model.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+
+    # TODO (tym): Figure out how to switch between LBANN builds. See
+    # GitHub Issue #1289.
```
[the remark this answers is not in the document]
Comment (timmoon10) [reply bamboo/integration_tests/test_integration_resnet18.py]: I think this is also a problem in the unit tests (#1289).

[the remark this answers is not in the document]
Comment (forsyth2) [reply bamboo/integration_tests/test_integration_resnet18.py]: Ah ok then. So we'll just leave it testing one compiler and then have another PR to fix #1289 on both integration and unit tests?

File: bamboo/integration_tests/test_integration_resnet18.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (forsyth2) [reply bamboo/integration_tests/test_integration_resnet18.py]: Ah ok, makes sense.

File: bamboo/integration_tests/common_code.py (hunk 1 above)
Comment (forsyth2) [reply bamboo/integration_tests/common_code.py]: If you think we don't *need* the autoencoder and non-vision models tested then I suppose just these two tests are good then.

File: bamboo/integration_tests/test_integration_resnet18.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (timmoon10) [reply bamboo/integration_tests/test_integration_resnet18.py]: Yep. I'd say the right answer is the modify the test infrastructure so it runs in the build directory. That way we don't need to worry about switching between builds.

File: bamboo/integration_tests/test_integration_resnet18.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+imagenet_fraction = 0.0009992   # Train with 1280 out of 1.28M samples
+
+# Top-1 classificaiton accuracy
+expected_train_accuracy_range = (1.5, 3.5)
+
+# Average mini-batch time (in sec) for each LC system
+expected_mini_batch_times = {
+    'pascal': 0.15,
+    'catalyst': 6.0,
+    'lassen': -1,
```
[the remark this answers is not in the document]
Comment (timmoon10) [reply bamboo/integration_tests/test_integration_resnet18.py]: Removed the `-1`s.

**Comment by timmoon10** [comment]
The two tests pass in [my Bamboo build](https://lc.llnl.gov/bamboo/browse/LBANN-TIM222-8).

File: bamboo/integration_tests/common_code.py (hunk 1 above)
Comment (bvanessen) [reply bamboo/integration_tests/common_code.py]: Just jumping in here but we are going to want non-vision models tested and autoencoders.  Ideally, we will have integrated tests for every LBANN application.  Time will be an issue here.

**Pull request closed without merging**